folium/raster_layers.py

```diff
@@ -104,14 +106,26 @@ def __init__(
         subdomains: str = "abc",
         tms: bool = False,
         opacity: float = 1,
-        **kwargs
+        **kwargs,
     ):
-        # check for xyzservices.TileProvider without importing it
-        if isinstance(tiles, dict):
+        if isinstance(tiles, str):
+            if tiles.lower() == "openstreetmap":
+                tiles = "OpenStreetMap Mapnik"
+                if name is None:
+                    name = "openstreetmap"
+            try:
+                tiles = xyzservices.providers.query_name(tiles)
+            except ValueError:
+                # no match, likely a custom URL
+                pass
+
+        if isinstance(tiles, xyzservices.TileProvider):
             attr = attr if attr else tiles.html_attribution  # type: ignore
             min_zoom = tiles.get("min_zoom", min_zoom)
             max_zoom = tiles.get("max_zoom", max_zoom)
             subdomains = tiles.get("subdomains", subdomains)
+            if name is None:
+                name = tiles.name.replace(".", "").lower()
```

Comment:
are you following Folium convention here or is this another convention to create a name like this? I wonder if this is a chance to have a better name where we don't mush all words together in lower case. What do you think?

Reply:
I was just ensuring the name remains the same it was. I am happy to use proper name that comes from xyzservices, i.e. `"CartoDB Positron` if backwards compatibility of this particular bit is not important.

Reply:
Good to know. I don't really have an overview of what the impact of changing the name format would be. This is used as JS object name as well if I'm not mistaken, so maybe better to not change it lightly. Let's indeed keep it then as it is, we can look into improving that later.